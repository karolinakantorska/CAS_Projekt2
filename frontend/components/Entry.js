import React, { useState, useEffect } from "react";
import styled from 'styled-components';

const Entry = (props) => {
    const {time, userName, userEmail, id} = props;
return (
    <EntrySpan className='entryContainer'>
        <div className={time}>
            <p>gast name: ${userName}</p>
            <p>gast email: ${userEmail}</p>
        </div>
    </EntrySpan>
);
}
const EntrySpan = styled.span`
    background: white;
    font-size: 0.6rem;
    & .AM {
      background: lightskyblue;
      border: 1px solid gray;
      padding: 0;
    }
    & .PM {
      background: powderblue;
      border: 1px solid gray;
    }
    & .AM::before {
      content: "AM"
    }
    & .PM::before {
      content: "PM"
    }
`;

export default Entry;