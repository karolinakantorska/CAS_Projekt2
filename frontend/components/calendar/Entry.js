import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Entry = (props) => {
  const {
    id,
    time,
    userName,
    userEmail,
    currentUserPermission,
  } = props;

  return (
    <EntrySpan className="entryContainer">
      <div className={time}>
        {currentUserPermission === 'USER' && <p>already booked!</p>}
        {(currentUserPermission === 'ADMIN' ||
          currentUserPermission === 'GUIDE') && (
          <Link
            href={{
              pathname: '/info_booking',
              query: {
                userName,
                time,
                id,
              },
            }}
          >
            <span>
              <p>gast name: {userName}</p>
              <p>gast email: {userEmail}</p>
            </span>
          </Link>
        )}
      </div>
    </EntrySpan>
  );
};
Entry.PropTypes = {
  id: PropTypes.string,
  time: PropTypes.string,
  userName: PropTypes.string,
  userEmail: PropTypes.string,
  currentUserPermission: PropTypes.string,
};
const EntrySpan = styled.span`
  background: white;
  font-size: 0.6rem;
  & .AM {
    background: lightskyblue;
    border: 1px solid gray;
    padding: 0;
  }
  & .PM {
    background: powderblue;
    border: 1px solid gray;
  }
  & .DAY {
    background: mistyrose;
    border: 1px solid gray;
  }
  & .AM::before {
    content: 'AM';
  }
  & .PM::before {
    content: 'PM';
  }
  & .DAY::before {
    content: 'DAY';
  }
`;

export default Entry;
