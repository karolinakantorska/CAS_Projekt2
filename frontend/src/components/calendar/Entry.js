import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { permission } from '../../lib/utils';
import { routeToEditEntry } from '../../lib/utilsRouts';
// Components for Styling
import { StyledTextBody2 } from '../styles/StyledText';

const Entry = ({ reservation, currentUser }) => {
  function handleEntrySpanClick() {
    routeToEditEntry(reservation.id);
  }
  if (currentUser.permissions === permission.user) {
    return (
      <EntrySpan className={reservation.time}>
        {/* TODO Ids would be better*/}
        {reservation.userEmail === currentUser.email ? (
          <span className="your_booking">Your booking!</span>
        ) : (
          <span>Booked!</span>
        )}
      </EntrySpan>
    );
  }
  if (currentUser.permissions === permission.admin) {
    return (
      <EntrySpan className={reservation.time} onClick={() => handleEntrySpanClick()}>
        <StyledTextBody2>{`Gast: ${reservation.userName}`}</StyledTextBody2>
        <StyledTextBody2>{`Guide: ${reservation.guide.name}`}</StyledTextBody2>
      </EntrySpan>
    );
  }
  if (currentUser.permissions === permission.guide) {
    return (
      <EntrySpan className={reservation.time} onClick={() => handleEntrySpanClick()}>
        {/* TODO Ids would be better*/}
        {reservation.userEmail === reservation.guide.email ? (
          <StyledTextBody2 className="holiday">Free!</StyledTextBody2>
        ) : (
          <StyledTextBody2>{`Gast: ${reservation.userName}`}</StyledTextBody2>
        )}
      </EntrySpan>
    );
  } else {
    return <p>booked</p>;
  }
};
Entry.propTypes = {
  reservation: PropTypes.object,
  currentUser: PropTypes.object,
};
const EntrySpan = styled.span`
  display: grid;
  //grid-template-rows: 1fr 1fr;
  max-width: 138px;
  font-size: 0.9rem;
  margin-top: 6px;
  border-radius: 5px;
  background: rgba(217, 217, 217, 0.5);
  align-content: center;
  justify-content: center;
  .holiday {
    color: red;
  }
  .your_booking {
    color: red;
  }
  .AM::before {
    content: 'AM: ';
  }
  .PM::before {
    content: 'PM: ';
  }
  .DAY::before {
    content: 'DAY: ';
  }
`;

export default Entry;
