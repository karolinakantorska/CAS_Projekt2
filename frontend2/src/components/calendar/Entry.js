import React from 'react';
import PropTypes from 'prop-types';
//Components
import EntryGuide from '../booking/EntryGuide';
import EntryAdmin from '../booking/EntryAdmin';
import EntryUser from '../booking/EntryUser';
import { permission } from '../../lib/utils';

const Entry = ({ reservation, currentUser }) => {
  if (currentUser.permissions === permission.user) {
    return <EntryUser reservation={reservation} currentUser={currentUser} />;
  }
  if (currentUser.permissions === permission.admin) {
    return <EntryAdmin reservation={reservation} />;
  }
  if (currentUser.permissions === permission.guide) {
    return <EntryGuide reservation={reservation} currentUser={currentUser} />;
  } else {
    return <p>booked</p>;
  }
};
Entry.propTypes = {
  reservation: PropTypes.object,
  currentUser: PropTypes.object,
};
export default Entry;
