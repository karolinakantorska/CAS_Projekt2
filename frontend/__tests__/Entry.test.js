import Entry from '../src/components/calendar/Entry';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

const id = '1';
const time = 'AM';
const userName = 'Michal';
const userEmail = 'michal@gmail.com';

describe('<Entry/>', () => {
  it('renders', () => {
    shallow(<Entry />);
  });
  it('matches the schnapshot Guide', () => {
    const wrapper = shallow(
      <Entry id={id} time={time} userName={userName} currentUserPermission={'USER'} />,
    );
    console.log(wrapper.debug());
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('matches the schnapshot User', () => {
    const wrapper = shallow(
      <Entry id={id} time={time} userName={userName} currentUserPermission={'GUIDE'} />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
