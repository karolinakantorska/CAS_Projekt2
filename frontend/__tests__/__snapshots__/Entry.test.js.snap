// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Entry/> matches the schnapshot Guide 1`] = `
<Entry__EntrySpan
  className="AM"
  onClick={[Function]}
>
  <div
    className="AM  grid_column_div"
  >
    <span>
      Booked!
    </span>
  </div>
</Entry__EntrySpan>
`;

exports[`<Entry/> matches the schnapshot User 1`] = `
<Entry__EntrySpan
  className="AM"
  onClick={[Function]}
>
  <div
    clasName="grid_row_div"
  >
    <StyledText__StyledTextBody2>
      Gast:
    </StyledText__StyledTextBody2>
    <StyledText__StyledTextBody2>
      Michal
    </StyledText__StyledTextBody2>
  </div>
</Entry__EntrySpan>
`;
